pkg/controller/certificates/internal/secretsmanager/secret.go

```diff
@@ -120,6 +123,40 @@ func (s *SecretsManager) UpdateData(ctx context.Context, crt *cmapi.Certificate,
 	return err
 }
 
+func updateSecretWithAdditionalOutputFormats(crt *cmapi.Certificate, secret *corev1.Secret, data SecretData) error {
+	if crt.Spec.AdditionalOutputFormats != nil {
+		additionalOutputFormatDER := false
+		additionalOutputFormatPEM := false
+		for _, f := range crt.Spec.AdditionalOutputFormats {
+			switch f.Type {
+			case cmapi.OutputFormatDER:
+				additionalOutputFormatDER = true
+			case cmapi.OutputFormatPEM:
+				additionalOutputFormatPEM = true
+			default:
+				return fmt.Errorf("Unknown additional output format %s", f.Type)
+			}
+		}
+		if additionalOutputFormatDER {
+			// Store binary format of the private key
+			block, _ := pem.Decode(data.PrivateKey)
+			secret.Data[cmapi.OutputFormatDERKey] = block.Bytes
+		} else {
+			delete(secret.Data, cmapi.OutputFormatDERKey)
+		}
+		if additionalOutputFormatPEM {
+			// Combine tls.key and tls.crt
+			secret.Data[cmapi.OutputFormatPEMKey] = []byte(strings.Join([]string{string(data.PrivateKey), string(data.Certificate)}, "\n"))
+		} else {
+			delete(secret.Data, cmapi.OutputFormatPEMKey)
+		}
+	} else {
+		delete(secret.Data, cmapi.OutputFormatDERKey)
+		delete(secret.Data, cmapi.OutputFormatPEMKey)
+	}
+	return nil
+}
```

Comment:
suggestion: I'd usually prefer to exit early if `crt.Spec.AdditionalOutputFormats == nil` and I think the code is more readable and idiomatic that way.

```suggestion
func updateSecretWithAdditionalOutputFormats(crt *cmapi.Certificate, secret *corev1.Secret, data SecretData) error {
	if crt.Spec.AdditionalOutputFormats == nil {
		delete(secret.Data, cmapi.OutputFormatDERKey)
		delete(secret.Data, cmapi.OutputFormatPEMKey)
		return nil
	}

	additionalOutputFormatDER := false
	additionalOutputFormatPEM := false

	for _, f := range crt.Spec.AdditionalOutputFormats {
		switch f.Type {
		case cmapi.OutputFormatDER:
			additionalOutputFormatDER = true
		case cmapi.OutputFormatPEM:
			additionalOutputFormatPEM = true
		default:
			return fmt.Errorf("Unknown additional output format %s", f.Type)
		}
	}

	if additionalOutputFormatDER {
		// Store binary format of the private key
		block, _ := pem.Decode(data.PrivateKey)
		secret.Data[cmapi.OutputFormatDERKey] = block.Bytes
	} else {
		delete(secret.Data, cmapi.OutputFormatDERKey)
	}

	if additionalOutputFormatPEM {
		// Combine tls.key and tls.crt
		secret.Data[cmapi.OutputFormatPEMKey] = []byte(strings.Join([]string{string(data.PrivateKey), string(data.Certificate)}, "\n"))
	} else {
		delete(secret.Data, cmapi.OutputFormatPEMKey)
	}

	return nil
}
```

Reply:
ok, I've exited early like suggested